core/controllers/blog_dashboard.py

```diff
@@ -353,3 +354,86 @@ def delete(self, blog_post_id: str) -> None:
         blog_domain.BlogPost.require_valid_blog_post_id(blog_post_id)
         blog_services.delete_blog_post(blog_post_id)
         self.render_json(self.values)
+
+
+class BlogDashboardBlogPostStatisticsHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Handler for blog dashboard statistics tab for blog post."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'blog_post_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        },
+        'chart_type': {
+            'schema': {
+                'type': 'basestring',
+                'choices': [
+                    'views',
+                    'reads',
+                    'reading_time'
+                ]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {
+        'GET': {}
+    }
+
+    @acl_decorators.can_access_blog_dashboard
+    def get(self, blog_post_id: str, chart_type: str) -> None:
+        """Populates the data for generating statistics plot."""
+        stats_dict = (
+            blog_statistics_services.get_blog_post_stats_by_blog_post_id(
+                blog_post_id, chart_type
+            ).to_frontend_dict()
+        )
+        self.values.update({
+            'chart_type': chart_type,
+            'stats': stats_dict
+        })
+
+        self.render_json(self.values)
+
+
+class BlogDashboardAuthorBlogPostsStatisticsHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Handler for blog dashboard statistics tab for author"""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'chart_type': {
+            'schema': {
+                'type': 'basestring',
+                'choices': [
+                    'views',
+                    'reads',
+                    'reading_time'
+                ]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {
+        'GET': {}
+    }
+
+    @acl_decorators.can_access_blog_dashboard
+    def get(self, chart_type: str) -> None:
+        """Populates the data for generating author statistics plot."""
+        author_id = self.user_id
+        assert author_id is not None
```

Comment:
```suggestion
        assert self.user_id is not None
        author_id = self.user_id
```

Because assert is only allowed for `self.user_id`.

Reply:
Done